x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -292,7 +292,22 @@ export const useInitSourcerer = (
 };
 
 const LOGS_WILDCARD_INDEX = 'logs-*';
-export const EXCLUDE_ELASTIC_CLOUD_INDEX = '-*elastic-cloud-logs-*';
+export const EXCLUDE_ELASTIC_CLOUD_INDICES = [
+  '-*elastic-cloud-logs-*',
+  '-.ds-logs-elastic_agent*',
+  '-.ds-logs-system-*',
+];
+
+const sortWithExcludesAtEnd = (indices: string[]) => {
+  const allSorted = indices.reduce(
+    (acc: { includes: string[]; excludes: string[] }, index) =>
+      index.split('').shift() === '-'
+        ? { includes: acc.includes.sort(), excludes: [...acc.excludes, index].sort() }
+        : { includes: [...acc.includes, index].sort(), excludes: acc.excludes.sort() },
+    { includes: [], excludes: [] }
+  );
+  return [...allSorted.includes, ...allSorted.excludes];
```

Comment:
Consider simplifying from:

```ts
  const allSorted = indices.reduce(
    (acc: { includes: string[]; excludes: string[] }, index) =>
      index.split('').shift() === '-'
        ? { includes: acc.includes.sort(), excludes: [...acc.excludes, index].sort() }
        : { includes: [...acc.includes, index].sort(), excludes: acc.excludes.sort() },
    { includes: [], excludes: [] }
  );
  return [...allSorted.includes, ...allSorted.excludes];
```

to something like:

```ts
  const includes = indices.filter((x) => !x.trim().startsWith('-'));
  const excludes = indices.filter((x) => x.trim().startsWith('-'));

  return [...includes.sort(), ...excludes.sort()];
```

Reply:
ok i did part of this but left the reduce